### PR TITLE
2022 Q2 release

### PR DESCRIPTION
**Pull request by svrooij**
- Cli framework migration
- Mute/unmute and toggle mute commands
- Clear queue command
- Next / Previous command
- Play command now also switches to the queue (not sure if this is a good idea)

Replaced #17

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -278,16 +283,49 @@ OPTIONS
...
+  -a, --available        Install a specific version.
+  -i, --interactive      Interactively select version to install. This is ignored if a channel is provided.
+  -v, --version=version  Install a specific version.
+  --force                Force a re-download of the requested version.
+
+EXAMPLES
+  [object Object]
+  [object Object]
+  [object Object]
+  [object Object]
```
Comment (mdouglass) [inline README.md]: Guessing this is unintentional?

**Comment by mdouglass** [comment]
The handling of missing command line arguments isn't as nice as in the previous oclif version:

old behavior:
```
$ sonos info
 ›   Error: Missing 2 required args:
 ›   device  Name or uuid of player
 ›   kind    What do you want to load
 ›   See more help with --help
```

new behavior:
```
$ sonos info                                                
Error: Missing 2 required args:
device  Name or uuid of player
kind    What do you want to load
See more help with --help
    at validateArgs (/home/matthew/projects/sonos-cli-upstream/node_modules/@oclif/core/lib/parser/validate.js:29:19)
    at Object.validate (/home/matthew/projects/sonos-cli-upstream/node_modules/@oclif/core/lib/parser/validate.js:79:5)
    at Object.parse (/home/matthew/projects/sonos-cli-upstream/node_modules/@oclif/core/lib/parser/index.js:31:7)
    at async Info.run (/home/matthew/projects/sonos-cli-upstream/src/commands/info.ts:18:27)
    at async Info._run (/home/matthew/projects/sonos-cli-upstream/node_modules/@oclif/core/lib/command.js:67:22)
    at async Config.runCommand (/home/matthew/projects/sonos-cli-upstream/node_modules/@oclif/core/lib/config/config.js:268:25)
    at async Object.run (/home/matthew/projects/sonos-cli-upstream/node_modules/@oclif/core/lib/main.js:78:5)
```

**Comment by mdouglass** [comment]
`bin/dev` should be marked as an executable file

**Comment by mdouglass** [comment]
Ok, other than those three minor things, looks like everything else is gtg. Thanks!

**Comment by svrooij** [comment]
1. I'm on Windows, no clue how to mark a file as executable 😊 
2. Auto update removed (ugly `object.....` in Readme), wasn't using auto update anyway.
3. I think the stack trace is by design, see:

image

If you agree, I'll continue and merge this pull request. Planning a more regular release schedule.

**Comment by mdouglass** [comment]
That all looks great.  I can send a one-liner PR shortly that marks dev as executable - but since it's only for internal use, there's no reason for it to hold this one up.

**Comment by mdouglass** [comment]
added https://github.com/svrooij/sonos-cli/pull/20 against mdouglass/master which just marks dev executable

**Pull request merged**